packages/devtools_app/test/association_variable_test.dart

```diff
@@ -0,0 +1,184 @@
+      return instance;
+    });
+
+    await debuggerController.buildVariablesTree(variable);
+
+    expect(variable.children, [
+      matchesVariable(name: '[\'Hey\']', value: '12.34'),
+    ]);
+  });
+
+  test('Creates bound variables for Map with Int key and Double value',
+      () async {
+    final instance = Instance(
+      kind: InstanceKind.kMap,
+      id: '123',
+      classRef: null,
+      associations: [
+        MapAssociation(
+          key: InstanceRef(
+              classRef: null,
+              id: '4',
+              kind: InstanceKind.kInt,
+              valueAsString: '1'),
+          value: InstanceRef(
+            classRef: null,
+            id: '5',
+            kind: InstanceKind.kDouble,
+            valueAsString: '12.34',
+          ),
+        ),
+      ],
+    );
+    final variable = Variable.create(BoundVariable(
+      name: 'test',
+      value: instance,
+      declarationTokenPos: null,
+      scopeEndTokenPos: null,
+      scopeStartTokenPos: null,
+    ));
+    when(manager.service.getObject(any, any)).thenAnswer((_) async {
+      return instance;
+    });
+
+    await debuggerController.buildVariablesTree(variable);
+
+    expect(variable.children, [
+      matchesVariable(name: '[1]', value: '12.34'),
+    ]);
+  });
+
+  test('Creates bound variables for Map with Object key and Double value',
+      () async {
+    final instance = Instance(
+      kind: InstanceKind.kMap,
+      id: '123',
+      classRef: null,
+      associations: [
+        MapAssociation(
+          key: InstanceRef(
+            classRef: ClassRef(id: 'a', name: 'Foo'),
+            id: '4',
+            kind: InstanceKind.kPlainInstance,
+          ),
+          value: InstanceRef(
+            classRef: null,
+            id: '5',
+            kind: InstanceKind.kDouble,
+            valueAsString: '12.34',
+          ),
+        ),
+      ],
+    );
+    final variable = Variable.create(BoundVariable(
+      name: 'test',
+      value: instance,
+      declarationTokenPos: null,
+      scopeEndTokenPos: null,
+      scopeStartTokenPos: null,
+    ));
+    when(manager.service.getObject(any, any)).thenAnswer((_) async {
+      return instance;
+    });
+
+    await debuggerController.buildVariablesTree(variable);
+
+    expect(variable.children, [
+      matchesVariable(name: '[key]', value: 'Foo'),
+      matchesVariable(name: '[value]', value: '12.34'),
+    ]);
+  });
+}
+
+Matcher matchesVariable({
+  @required String name,
+  @required Object value,
+}) {
+  return const TypeMatcher<Variable>()
+      .having(
+        (v) => v.displayValue,
+        'displayValue',
+        equals(value),
+      )
+      .having(
+          (v) => v.boundVar,
+          'boundVar',
+          const TypeMatcher<BoundVariable>()
+              .having((bv) => bv.name, 'name', equals(name)));
+}
+
+class MockVmServiceWrapper extends Mock implements VmServiceWrapper {}
```

Comment:
we already have a mock for VmServiceWrapper. See `MockVmService` in support/mocks.dart. If we can use FakeServiceManager, the mock vm service is included by default.

Reply:
Done